manual/parts/vspace.tex

```diff
@@ -109,22 +109,19 @@ \subsubsection{AArch32}
 \subsubsection{AArch64}
 
 Depending on configuration, Arm AArch64 processors have page-table structures with 3 or 4 levels.
-The VSpace object is therefore implemented either by the \obj{PageGlobalDirectory} object (4 level
-configs) or the \obj{PageUpperDirectory} object (3 level configs). To help with writing code that is
-generic in this distinction, \texttt{libsel4} provides the macro \texttt{seL4\_ARM\_VSpaceObject}
-that expands to the correct corresponding object. All intermediate paging structures are indexed by
-9 bits of the virtual address. Depending on configuration, the top-level object is indexed by either
-9 or 10 bits. The macro \texttt{seL4\_VSpaceIndexBits} makes this value available under a generic
-name. The table below shows the four-level configuration. On three-level configurations, the
-invocations for \texttt{seL4\_ARM\_VSpaceObject} capabilities are instead available on
-\obj{PageUpperDirectory} caps.
+Regardless of the configuration, the VSpace object is implemented by the \obj{PageTable} object.
+To help with writing code that is generic, \texttt{libsel4} provides the macro \texttt{seL4\_ARM\_VSpaceObject}
+that expands to the correct corresponding object across architectures. All intermediate paging
```

Comment:
It might be better to just omit the mention of `seL4\_ARM\_VSpaceObject`, RISC-V doesn't mention it either.

Reply:
How else are you supposed to find out about it?

Reply:
I was trying to say that there is no need for `seL4_ARM_VSpaceObject` now that the top-level page table object is always the same on AArch64.

Reply:
and if there is no need for it, why should the manual mention it?

Reply:
Sorry for the drive-by comment, I haven't had time to read the full change yet, but  `Regardless of the configuration, the VSpace object is implemented by the \obj{PageTable} object` is not true for AArch64 -- depending on config these can be different objects and object sizes.

Reply:
What other objects is a VSpace represented by?

Reply:
Also what I said here with: "expands to the correct corresponding object across architectures" is wrong obviously since `seL4_ARM_VSpaceObject` is specific to ARM.

Reply:
> What other objects is a VSpace represented by?

VSpaceObject is its own type in AArch64, mostly the same but not quite the same as a PageTableObject. There are other operations available on it and its size can be different from a PageTableObject.

Reply:
Okay yep, thanks for clearing up my confusion. The thing I don't understand is why the current manual says the following: `The VSpace object is therefore implemented either by the \obj{PageGlobalDirectory} object (4 level
configs) or the \obj{PageUpperDirectory} object (3 level configs)`. I was just trying to adjust this sentence with this PR to remove PUD/PGD mentions.

Should we just remove this sentence to avoid confusing an actual VSpace object with what it is "implemented by"?

Reply:
I think it's good to mention that the root page table is special and a different object, namely `VSpaceObject` and not a `PageTableObject`. And describing the page table layout options for Aarch64 is useful too. What about:

```
Depending on configuration, Arm AArch64 processors have page-table structures with 3 or 4 levels.
The VSpace object is \texttt{seL4\_ARM\_VSpaceObject}, which is a distinct object type used for the 
top level page table. All intermediate paging structures are indexed by 9 bits of the virtual address 
and  are \obj{PageTable} objects. Depending on configuration, the top-level object is indexed by either 
9 or 10 bits. The macro \texttt{seL4\_VSpaceIndexBits} makes this value available under a generic name.
The table below shows the four-level configuration.
```

Reply:
Thanks, I've adjusted the wording to this.